python/test_infra/ttir/add.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from typing import Optional, List
+
+from .ttir_op_test_base import TTIROpTestBase
+
+from ttmlir.ir import Value, Context, Location, OpView
+from ttmlir.dialects import ttir, tt
+
+
+class Add(TTIROpTestBase):
+    """Test harness for `ttir.add` op."""
+
+    def __init__(self, ctx: Context, location: Location):
+        super().__init__(ctx, location)
+
+    @property
+    def name(self) -> str:
+        return "add"
+
+    @property
+    def num_inputs(self) -> int:
+        return 2
+
+    @property
+    def num_outputs(self) -> int:
+        return 1
```

Comment:
Can the above 3 interface functions just be arguments to the top level constructor?

```python
super().__init__(ctx, location, name="add", num_inputs=2, num_outputs=1)
```

Reply:
Makes sense.